src/compiler/checker.ts

```diff
@@ -6551,19 +6609,30 @@ module ts {
             }
         }
 
+        function createPromiseType(awaitedType: Type, func: FunctionLikeDeclaration): Type {
+            if (globalPromiseSymbol) {
+                awaitedType = getAwaitedType(awaitedType, awaitedType);
+                var promiseType = globalPromiseType || getDeclaredTypeOfSymbol(globalPromiseSymbol);
+                return promiseType !== emptyObjectType ? createTypeReference(<GenericType>promiseType, [awaitedType]) : emptyObjectType;

```

Comment:
This follows roughly the same logic as `createArrayType`. If you are using a lib.d.ts that does not have a global `Promise` type defined, `getDeclaredTypeOfSymbol` will report an error and return `emptyObjectType`.

From: Jason Freeman [mailto:notifications@github.com]
Sent: Thursday, February 12, 2015 3:51 PM
To: Microsoft/TypeScript
Cc: Ron Buckton
Subject: Re: [TypeScript] Async Functions (#1781)


In src/compiler/checker.ts<https://github.com/Microsoft/TypeScript/pull/1781#discussion_r24633034>:

> @@ -6551,19 +6609,30 @@ module ts {

>              }

>          }

>

> +        function createPromiseType(awaitedType: Type, func: FunctionLikeDeclaration): Type {

> +            if (globalPromiseSymbol) {

> +                awaitedType = getAwaitedType(awaitedType, awaitedType);

> +                var promiseType = globalPromiseType || getDeclaredTypeOfSymbol(globalPromiseSymbol);

> +                return promiseType !== emptyObjectType ? createTypeReference(<GenericType>promiseType, [awaitedType]) : emptyObjectType;

Why would promiseType equal emptyObjectType

—
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/TypeScript/pull/1781/files#r24633034>.